src/board.cpp

```
@@ -0,0 +1,113 @@
+
+void Board::add_cell(CellularAutomaton *cell) {
+    if (type == CONTINUOUS) {
+        auto *continuous_cell = dynamic_cast<ContinuousAutomaton*>(cell);
+        if (continuous_cell && current_population < population) {
+            continuous[current_population] = continuous_cell;
+            current_population++;
+        }
+    }
+    else if (type == GRID) {
+        auto *discrete_cell = dynamic_cast<DiscreteAutomaton*>(cell);
+        if (discrete_cell && discrete_cell->get_x() < width && discrete_cell->get_y() < height) {
```

cppcheck

[performance] Variable 'species' is assigned in constructor body. Consider performing initialization in initialization list.

(useInitializationList)